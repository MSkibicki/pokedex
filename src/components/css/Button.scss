@import "./Global.scss";

.button {
  display: flex;
  justify-content: center;
  margin-bottom: 3em;
  margin-top: 1em;

  .button-back {
    height: 40px;
    width: 150px;
    border-radius: 10px;
    margin-right: 1em;
    background: linear-gradient(120deg, $white, $black);
    color: $main-color;
  }

  .button-back:hover {
    cursor: pointer;
    box-shadow: 5px 5px 5px $black;
    outline: 0;
  }
}
