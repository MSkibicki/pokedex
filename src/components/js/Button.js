import React from "react";
import { Link } from "react-router-dom";
import "../css/Button.scss";

const Button = ({ text }) => {
  return (
    <div className="button">
      <Link to={"/"}>
        <button className="button-back">{text}</button>
      </Link>
    </div>
  );
};

export default Button;
