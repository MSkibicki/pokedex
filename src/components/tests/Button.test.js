import Button from "./../js/Button";

describe("<Button />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Button text="Back to main page" />);

    expect(wrapper).toMatchSnapshot();
  });

  it("displays button text correctly", () => {
    const wrapper = shallow(<Button text="Back to main page" />);

    const text = wrapper.find(".button-back").text();

    expect(text).toEqual("Back to main page");
  });

  it("has correct div classname", () => {
    const wrapper = shallow(<Button text="Back to main page" />);

    const divClass = wrapper.find("div").hasClass("button");

    expect(divClass).toBe(true);
  });

  it("has correct button classname", () => {
    const wrapper = shallow(<Button text="Back to main page" />);

    const buttonClass = wrapper.find("button").hasClass("button-back");

    expect(buttonClass).toBe(true);
  });
});
