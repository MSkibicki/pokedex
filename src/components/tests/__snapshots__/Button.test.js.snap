// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<div
  className="button"
>
  <Link
    to="/"
  >
    <button
      className="button-back"
    >
      Back to main page
    </button>
  </Link>
</div>
`;
